src/main/java/fruitshop/service/impl/ReportGeneratorImpl.java

```diff
@@ -0,0 +1,22 @@
+package fruitshop.service.impl;
+
+import fruitshop.db.Storage;
+import fruitshop.service.ReportGenerator;
+import java.util.Map;
+
+public class ReportGeneratorImpl implements ReportGenerator {
+    public static final String HEADER = "fruit,quantity";
+    public static final String COMA = ",";
+    private final Storage storage = new Storage();
```

Comment:
Instantiating a new `Storage` object within the `ReportGeneratorImpl` violates the SOLID principles. Consider passing the `Storage` instance to the `ReportGeneratorImpl` through its constructor, making it more flexible and testable.